### ISSUE TITLE
Rasa restarts session while filling slots with a form and deletes slots

### ISSUE THREAD

**Issue by mexophantes** [RasaHQ/rasa#5091]
**Rasa version**:
1.6.1

**Rasa SDK version** (if used & relevant):
1.6.1

**Rasa X version** (if used & relevant):
0.24.1

**Python version**:
Python 3.6.10 :: Anaconda, Inc.
Tested also with Python 3.7 --> same result

**Operating system** (windows, osx, ...):
windows

**Issue**:
Rasa restarts session (calls action_session_start) while filling slots with a form and deletes slots.
See also forum: https://forum.rasa.com/t/slots-getting-reset-too-early-when-using-sessions/23559

**Error (including full traceback)**:
```
2020-01-20 09:43:24 DEBUG    rasa.core.policies.memoization  - There is a memorised next action '28'
2020-01-20 09:43:24 DEBUG    rasa.core.policies.form_policy  - There is no active form
2020-01-20 09:43:24 DEBUG    rasa.core.policies.mapping_policy  - There is no mapped action for the predicted intent, 'request_issue_category'.
2020-01-20 09:43:24 DEBUG    rasa.core.policies.ensemble  - Predicted next action using policy_1_MemoizationPolicy
2020-01-20 09:43:24 DEBUG    rasa.core.processor  - Predicted next action 'complaint_form' with confidence 1.00.
2020-01-20 09:43:24 DEBUG    rasa.core.actions.action  - Calling action endpoint to run action 'complaint_form'.
Kategorie "Öffentliche Ordnung", alles klar.
2020-01-20 09:43:24 DEBUG    rasa.core.processor  - Action 'complaint_form' ended with events '[BotUttered('Um was geht es?', {"elements": null, "quick_replies": null, "buttons": null, "attachment": null, "image": null, "custom": null}, {"category": "\u00d6ffentliche Ordnung", "first_name": "liebe Weltverbesserer und Weltverbessererinnen"}, 1579509804.3373435), <rasa.core.events.Form object at 0x000002A81918BC18>, <rasa.core.events.SlotSet object at 0x000002A81918B7F0>]'.
2020-01-20 09:43:24 DEBUG    rasa.core.processor  - Current slot values:
        additional_information: None
        category: Öffentliche Ordnung
        feedback: None
        first_name: liebe Weltverbesserer und Weltverbessererinnen
        image_filename: None
        last_name: None
        received: None
        requested_slot: what
        time: None
        what: None
        when: None
        where: None
        who: None
2020-01-20 09:43:24 DEBUG    rasa.core.policies.fallback  - NLU confidence threshold met, confidence of fallback action set to core threshold (0.3).
2020-01-20 09:43:24 DEBUG    rasa.core.policies.memoization  - Current tracker state [{'prev_utter_initial': 1.0, 'intent_greet': 1.0}, {'prev_action_listen': 1.0, 'intent_request_new_issue': 1.0}, {'prev_utter_issue_category': 1.0, 'intent_request_new_issue': 1.0}, {'prev_action_listen': 1.0, 'intent_request_issue_category': 1.0, 'entity_category': 1.0}, {'prev_utter_selected_issue_category': 1.0, 'intent_request_issue_category': 1.0, 'entity_category': 1.0}]
2020-01-20 09:43:24 DEBUG    rasa.core.policies.memoization  - There is a memorised next action '28'
2020-01-20 09:43:24 DEBUG    rasa.core.policies.form_policy  - There is an active form 'complaint_form'
2020-01-20 09:43:24 DEBUG    rasa.core.policies.mapping_policy  - There is no mapped action for the predicted intent, 'request_issue_category'.
2020-01-20 09:43:24 DEBUG    rasa.core.policies.ensemble  - Predicted next action using policy_2_FormPolicy
2020-01-20 09:43:24 DEBUG    rasa.core.processor  - Predicted next action 'action_listen' with confidence 1.00.
2020-01-20 09:43:24 DEBUG    rasa.core.processor  - Action 'action_listen' ended with events '[]'.
2020-01-20 09:43:24 DEBUG    rasa.core.tracker_store  - Tracker with sender_id 'default' stored to database
2020-01-20 09:43:24 DEBUG    rasa.core.lock_store  - Deleted lock for conversation 'default'.
Um was geht es?
Your input ->  Straßenlampe ist kaputt
2020-01-20 09:43:33 DEBUG    rasa.core.tracker_store  - Recreating tracker from sender id 'default'
2020-01-20 09:43:33 DEBUG    rasa.core.processor  - Starting a new session for conversation ID 'default'.
2020-01-20 09:43:33 DEBUG    rasa.core.processor  - Action 'action_session_start' ended with events '[<rasa.core.events.SessionStarted object at 0x000002A8192B44A8>, <rasa.core.events.ActionExecuted object at 0x000002A8192B44E0>]'.
2020-01-20 09:43:33 DEBUG    rasa.core.processor  - Current slot values:
        additional_information: None
        category: None
        feedback: None
        first_name: None
        image_filename: None
        last_name: None
        received: None
        requested_slot: None
        time: None
        what: None
        when: None
        where: None
        who: None
2020-01-20 09:43:34 DEBUG    rasa.core.tracker_store  - Tracker with sender_id 'default' stored to database
2020-01-20 09:43:34 WARNING  root  - Could not parse timestamp 8f2e26c3b3754873abc9dc10df626b65. Instead current UTC time will be passed to duckling. Error: invalid literal for int() with base 10: '8f2e26c3b3754873abc9dc10df626b65'
2020-01-20 09:43:34 DEBUG    rasa.core.processor  - Received user message 'Straßenlampe ist kaputt' with intent '{'name': 'inform', 'confidence': 0.592702099446641}' and entities '[{'start': 0, 'end': 12, 'value': 'straßenlampe', 'entity': 'what', 'confidence': 0.6089951002663254, 'extractor': 'CRFEntityExtractor'}]'
2020-01-20 09:43:34 DEBUG    rasa.core.processor  - Current slot values:
        additional_information: None
        category: None
        feedback: None
        first_name: None
        image_filename: None
        last_name: None
        received: None
        requested_slot: None
        time: None
        what: None
        when: None
        where: None
        who: None
2020-01-20 09:43:34 DEBUG    rasa.core.processor  - Logged UserUtterance - tracker now has 29 events.
2020-01-20 09:43:34 DEBUG    rasa.core.policies.fallback  - NLU confidence threshold met, confidence of fallback action set to core threshold (0.3).
2020-01-20 09:43:34 DEBUG    rasa.core.policies.memoization  - Current tracker state [None, None, None, {}, {'intent_inform': 1.0, 'entity_what': 1.0, 'prev_action_listen': 1.0}]
2020-01-20 09:43:34 DEBUG    rasa.core.policies.memoization  - There is no memorised next action
2020-01-20 09:43:34 DEBUG    rasa.core.policies.form_policy  - There is no active form
2020-01-20 09:43:34 DEBUG    rasa.core.policies.ensemble  - Predicted next action using policy_0_FallbackPolicy
2020-01-20 09:43:34 DEBUG    rasa.core.processor  - Predicted next action 'action_default_fallback' with confidence 0.30.
2020-01-20 09:43:34 DEBUG    rasa.core.processor  - Action 'action_default_fallback' ended with events '[<rasa.core.events.UserUtteranceReverted object at 0x000002A819266DD8>]'.
2020-01-20 09:43:34 DEBUG    rasa.core.processor  - Current slot values:
        additional_information: None
        category: None
        feedback: None
        first_name: None
        image_filename: None
        last_name: None
        received: None
        requested_slot: None
        time: None
        what: None
        when: None
        where: None
        who: None
2020-01-20 09:43:34 DEBUG    rasa.core.processor  - Predicted next action 'action_listen' with confidence 1.00.
2020-01-20 09:43:34 DEBUG    rasa.core.processor  - Action 'action_listen' ended with events '[]'.
2020-01-20 09:43:34 DEBUG    rasa.core.tracker_store  - Tracker with sender_id 'default' stored to database
2020-01-20 09:43:34 DEBUG    rasa.core.lock_store  - Deleted lock for conversation 'default'.
```

### Content of the tracker

| 2765 | default | action          | 1579510000 |                        | action_session_start                                         | {"event": "action",  "timestamp": 1579509795.0533154, "name":  "action_session_start", "policy": null,  "confidence": null} |
| ---- | ------- | --------------- | ---------- | ---------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
| 2766 | default | session_started | 1579510000 |                        |                                                              | {"event": "session_started", "timestamp":  1579509795.0533154} |
| 2767 | default | action          | 1579510000 |                        | action_listen                                                | {"event": "action", "timestamp":  1579509795.0533154, "name": "action_listen",  "policy": null, "confidence": null} |
| 2768 | default | user            | 1579510000 | greet                  |                                                              | {"event": "user", "timestamp":  1579509795.1603177, "text": "hi", "parse_data":  {"intent": {"name": "greet",  "confidence": 0.5941220507164725}, "entities": [],  "intent_ranking": [{"name": "greet",  "confidence": 0.5941220507164725}, {"name":  "affirm", "con... |
| 2769 | default | action          | 1579510000 |                        | action_get_username                                          | {"event": "action", "timestamp":  1579509795.2313168, "name": "action_get_username",  "policy": "policy_1_MemoizationPolicy",  "confidence": 1.0} |
| 2770 | default | slot            | 1579510000 |                        | first_name                                                   | {"event": "slot", "timestamp":  1579509795.2313168, "name": "first_name",  "value": "liebe Weltverbesserer und  Weltverbessererinnen"} |
| 2771 | default | action          | 1579510000 |                        | utter_initial                                                | {"event": "action", "timestamp":  1579509795.2593253, "name": "utter_initial",  "policy": "policy_1_MemoizationPolicy",  "confidence": 1.0} |
| 2772 | default | bot             | 1579510000 |                        |                                                              | {"event": "bot", "timestamp":  1579509795.2593253, "text": "Hallo liebe Weltverbesserer und  Weltverbessererinnen, willst du eine neue Beschwerde aufgeben oder die  Bestenliste sehen?", "data": {"elements": null,  "quick_replies": null, "buttons": [{"payload... |
| 2773 | default | action          | 1579510000 |                        | action_listen                                                | {"event": "action", "timestamp":  1579509795.2933195, "name": "action_listen",  "policy": "policy_1_MemoizationPolicy",  "confidence": 1.0} |
| 2774 | default | user            | 1579510000 | request_new_issue      | {"event": "user", "timestamp":  1579509800.030327, "text": "/request_new_issue",  "parse_data": {"text": "/request_new_issue",  "intent": {"name": "request_new_issue",  "confidence": 1.0}, "intent_ranking": [{"name":  "request_new_issue", "confidence": 1.0}],... |                                                              |
| 2775 | default | action          | 1579510000 |                        | utter_issue_category                                         | {"event": "action", "timestamp":  1579509800.0553136, "name": "utter_issue_category",  "policy": "policy_1_MemoizationPolicy",  "confidence": 1.0} |
| 2776 | default | bot             | 1579510000 |                        |                                                              | {"event": "bot", "timestamp":  1579509800.0553136, "text": "In welche Kategorie passt deine  Beschwerde? W\u00e4hle\n 1: \u00d6ffentliche Ordnung\n 2: Produkt\n 3: Im  Internet stimmt was nicht\n 4: Politik\n 5: Gesellschaft\n 6: Etwas  anderes", "data": {"e... |
| 2777 | default | action          | 1579510000 |                        | action_listen                                                | {"event": "action", "timestamp":  1579509800.0803158, "name": "action_listen",  "policy": "policy_1_MemoizationPolicy",  "confidence": 1.0} |
| 2778 | default | user            | 1579510000 | request_issue_category | {"event": "user", "timestamp":  1579509804.2413173, "text":  "/request_issue_category{\"category\":\"\u00d6ffentliche  Ordnung\"}", "parse_data": {"text":  "/request_issue_category{\"category\":\"\u00d6ffentliche  Ordnung\"}", "intent": {"name":  "request_issu... |                                                              |
| 2779 | default | slot            | 1579510000 |                        | category                                                     | {"event": "slot", "timestamp":  1579509804.2413173, "name": "category",  "value": "\u00d6ffentliche Ordnung"} |
| 2780 | default | action          | 1579510000 |                        | utter_selected_issue_category                                | {"event": "action", "timestamp":  1579509804.2713172, "name":  "utter_selected_issue_category", "policy":  "policy_1_MemoizationPolicy", "confidence": 1.0} |
| 2781 | default | bot             | 1579510000 |                        |                                                              | {"event": "bot", "timestamp":  1579509804.2713172, "text": "Kategorie \"\u00d6ffentliche  Ordnung\", alles klar.", "data": {"elements":  null, "quick_replies": null, "buttons": null,  "attachment": null, "image": null, "custom":  null}, "metadata": {}} |
| 2782 | default | action          | 1579510000 |                        | complaint_form                                               | {"event": "action", "timestamp":  1579509804.3433163, "name": "complaint_form",  "policy": "policy_1_MemoizationPolicy",  "confidence": 1.0} |
| 2783 | default | bot             | 1579510000 |                        |                                                              | {"event": "bot", "timestamp":  1579509804.3433163, "metadata": {"category":  "\u00d6ffentliche Ordnung", "first_name": "liebe  Weltverbesserer und Weltverbessererinnen"}, "text": "Um  was geht es?", "data": {"elements": null,  "quick_replies": null, "buttons"... |
| 2784 | default | form            | 1579510000 |                        | complaint_form                                               | {"event": "form", "timestamp":  1579509804.3433163, "name": "complaint_form"} |
| 2785 | default | slot            | 1579510000 |                        | requested_slot                                               | {"event": "slot", "timestamp":  1579509804.3433163, "name": "requested_slot",  "value": "what"} |
| 2786 | default | action          | 1579510000 |                        | action_listen                                                | {"event": "action", "timestamp":  1579509804.3723152, "name": "action_listen",  "policy": "policy_2_FormPolicy", "confidence":  1.0} |
| 2787 | default | action          | 1579510000 |                        | action_session_start                                         | {"event": "action", "timestamp":  1579509813.995316, "name": "action_session_start",  "policy": null, "confidence": null} |
| 2788 | default | session_started | 1579510000 |                        |                                                              | {"event": "session_started", "timestamp":  1579509813.995316} |
| 2789 | default | action          | 1579510000 |                        | action_listen                                                | {"event": "action", "timestamp":  1579509813.995316, "name": "action_listen",  "policy": null, "confidence": null} |
| 2790 | default | user            | 1579510000 | inform                 |                                                              | {"event": "user", "timestamp":  1579509814.059316, "text": "Stra\u00dfenlampe ist  kaputt", "parse_data": {"intent": {"name":  "inform", "confidence": 0.592702099446641},  "entities": [{"start": 0, "end": 12,  "value": "stra\u00dfenlampe", "entity":  "what", "... |
| 2791 | default | action          | 1579510000 |                        | action_default_fallback                                      | {"event": "action", "timestamp":  1579509814.0953343, "name": "action_default_fallback",  "policy": "policy_0_FallbackPolicy",  "confidence": 0.3} |
| 2792 | default | rewind          | 1579510000 |                        |                                                              | {"event": "rewind", "timestamp":  1579509814.0953343}        |
| 2793 | default | action          | 1579510000 |                        | action_listen                                                | {"event": "action", "timestamp":  1579509814.107341, "name": "action_listen",  "policy": null, "confidence": 1.0} |

**Command or request that led to error**:
`action_session_start` is called in function `_update_tracker_session` `processor.py` in `rasa\core` although session has not expired (session_expiration_time: 0.0) and events have been applied before.

```python
    async def _update_tracker_session(
        self, tracker: DialogueStateTracker, output_channel: OutputChannel,
    ) -> None:
        """Check the current session in `tracker` and update it if expired.

        An 'action_session_start' is run if the latest tracker session has expired,
        or if the tracker does not yet contain any events (only those after the last
        restart are considered).

        Args:
            tracker: Tracker to inspect.
            output_channel: Output channel for potential utterances in a custom
                `ActionSessionStart`.
        """
        if not tracker.applied_events() or self._has_session_expired(tracker):
            logger.debug(
                f"Starting a new session for conversation ID '{tracker.sender_id}'."
            )

            await self._run_action(
                action=self._get_action(ACTION_SESSION_START_NAME),
                tracker=tracker,
                output_channel=output_channel,
                nlg=self.nlg,
            )

            self.tracker_store.save(tracker)
```
**Content of configuration file (config.yml)** (if relevant):
```yml
language: de
pipeline:
- name: SpacyNLP
- name: SpacyTokenizer
- name: SpacyFeaturizer
- name: RegexFeaturizer
- name: CRFEntityExtractor
- name: DucklingHTTPExtractor
  url: http://localhost:32768
  dimensions:
  - time
  - number
  - amount-of-money
  - distance
  locale: de_DE
  timezone: Europe/Berlin
  timeout: 3
- name: EntitySynonymMapper
- name: SklearnIntentClassifier
policies:
- name: FallbackPolicy
- name: MemoizationPolicy
- name: FormPolicy
- name: MappingPolicy

```

**Content of domain file (domain.yml)** (if relevant):
```yml
session_config:
  session_expiration_time: 0.0
  carry_over_slots_to_new_session: true
intents:
- ...
entities:
- what
- when
- where
- who
slots:
  requested_slot:
    type: unfeaturized
  what:
    auto_fill: false 
    type: unfeaturized
  when:
    auto_fill: false
    type: unfeaturized
  where:
    auto_fill: false
    type: unfeaturized
  who:
    auto_fill: false
    type: unfeaturized
templates:
  utter_info:
  - text: ...
actions:
- ...
forms:
- complaint_form

```

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @amn41 will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by tmbo** [comment]
@ricwo probably best if you take a look at this

**Comment by wochinge** [comment]
This is actually a clear issue. The session expiration was disabled and the action was restarted anyway. That shouldn't happen.

**Comment by ricwo** [comment]
@mexophantes I'm not able to reproduce this on version 1.10.1. Could you please check if you still experience this?

**Comment by Govindaraju777** [comment]
Fallback response is working fine in rasa shell, getting expected utterance  : "Sorry, I didn't Understand, Try again..."(configured utterance utter_default)

But in rasa X UI getting response as "action_default_fallback(0.30)"
Any ides, why?

 My rasa installation details :
Rasa Version      :         2.8.2
Minimum Compatible Version: 2.8.0
Rasa SDK Version  :         2.8.1
Rasa X Version    :         0.39.3
Python Version    :         3.8.2


stories.yml:
- story: fallback
  steps:
    - intent: bot_challenge
    - action: action_default_fallback

rules.yml:
- rule: Ask the user to rephrase whenever they send a message with low NLU confidence
  steps:
  - intent: nlu_fallback
  - action: utter_please_rephrase

config.yml:

policies:
   - name: MemoizationPolicy
   - name: RulePolicy
   - name: FallbackPolicy
     nlu_threshold: 0.7
     nlu_core: 0.7
     core_fallback_threshold: 0.7
     fallback_action_name: "action_default_fallback"
     enable_fallback_prediction: True


domain.yml:
responses:
  utter_default:
  - text: Sorry I didn't get that. Can you rephrase?
